src/sentry/static/sentry/app/views/settings/organizationAuth/organizationAuthList.jsx

```diff
@@ -14,23 +14,45 @@ import getCookie from 'app/utils/getCookie';
 
 import ProviderItem from './providerItem';
 
-class OrganizationAuthList extends React.Component {
-  static contextTypes = {
-    organization: SentryTypes.Organization,
-  };
+const providerPopularity = {
+  google: 0,
+  github: 1,
+  okta: 2,
+  'active-directory': 3,
+  saml2: 4,
+  onelogin: 5,
+  rippling: 6,
+  auth0: 7,
+};
 
+class OrganizationAuthList extends React.Component {
   static propTypes = {
     providerList: PropTypes.arrayOf(SentryTypes.AuthProvider).isRequired,
     activeProvider: PropTypes.object,
   };
 
+  static contextTypes = {
+    organization: SentryTypes.Organization,
+  };
+
   render() {
     const {activeProvider} = this.props;
     const {organization} = this.context;
     const features = organization.features;
 
-    // Sort feature-flagged integrations last
-    const providerList = (this.props.providerList || []).sort((a, b) => {
+    // Sort provider list twice: first, by popularity,
```

Comment:
The double-sort here does sacrifice performance for readability. There'd be a miniscule performance improvement if we only sorted once, but the implementation of that (.reduce() to two arrays, one with feature enabled and one with feature disabled, and then sort each list by popularity and concat) is so much harder to grok that it's not worth it. And if a sort of ~10 elements ever becomes our perf bottleneck, I'll be so happy I'll re-write this section. :)

Reply:
There's <10 elements, so this has negligible effect.